### PR TITLE
KDE.Elisa version 23.04.00

### PR DESCRIPTION
**Pull request by sarvex**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/105917)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> Installation failed with exit code -2145844844 
> 2023-05-04 05:23:55.703 [CORE] DeliveryOptimization error: 0x80190194, extended error: 0x00000000 
> 2023-05-04 05:23:55.704 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\DODownloader.cpp(289)\WindowsPackageManager.dll!00007FF9D9321EFD: (caller: 00007FF9D93227D0) Exception(1) tid(d5c) 80190194 Not found (404). 
> 2023-05-04 05:23:59.285 [CORE] Download request failed. Returned status: 404 
> 2023-05-04 05:23:59.285 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Downloader.cpp(67)\WindowsPackageManager.dll!00007FF9D92E1D0B: (caller: 00007FF9D92E2129) Exception(2) tid(d5c) 80190194 Not found (404). 
> 2023-05-04 05:23:59.364 [CLI ] Failed to download, waiting a bit and retry. Url: https://binary-factory.kde.org/view/Windows%2064-bit/job/Elisa_Release_win64/lastSuccessfulBuild/artifact/elisa-23.04.0-1488-windows-cl-msvc2019-x86_64.exe 
> 2023-05-04 05:24:08.478 [CORE] DeliveryOptimization error: 0x80190194, extended error: 0x00000000 
> 2023-05-04 05:24:08.478 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\DODownloader.cpp(289)\WindowsPackageManager.dll!00007FF9D9321EFD: (caller: 00007FF9D93227D0) Exception(3) tid(d5c) 80190194 Not found (404). 
> 2023-05-04 05:24:11.212 [CORE] Download request failed. Returned status: 404 
> 2023-05-04 05:24:11.212 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Downloader.cpp(67)\WindowsPackageManager.dll!00007FF9D92E1D0B: (caller: 00007FF9D92E2129) Exception(4) tid(d5c) 80190194 Not found (404). 
> 2023-05-04 05:24:11.256 [CLI ] Caught wil::ResultException: D:\a\_work\1\s\external\pkg\src\AppInstallerCommonCore\Downloader.cpp(67)\WindowsPackageManager.dll!00007FF9D92E1D0B: (caller: 00007FF9D92E2129) Exception(4) tid(d5c) 80190194 Not found (404).

File: manifests/k/KDE/Elisa/23.04.00/KDE.Elisa.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: KDE.Elisa
+PackageVersion: 23.04.00
+Installers:
+- Architecture: x64
+  InstallerType: nullsoft
+  InstallerUrl: https://binary-factory.kde.org/view/Windows%2064-bit/job/Elisa_Release_win64/lastSuccessfulBuild/artifact/elisa-23.04.0-1488-windows-cl-msvc2019-x86_64.exe
```
Comment (stephengillie) [inline manifests/k/KDE/Elisa/23.04.00/KDE.Elisa.installer.yaml:9]:
```suggestion
  InstallerUrl: https://binary-factory.kde.org/view/Windows%2064-bit/job/Elisa_Release_win64/lastSuccessfulBuild/artifact/elisa-23.04.0-1489-windows-cl-msvc2019-x86_64.exe
```

Comment (stephengillie) [reply manifests/k/KDE/Elisa/23.04.00/KDE.Elisa.installer.yaml:9]: It appears that their build number incremented.

Comment (sarvex) [reply manifests/k/KDE/Elisa/23.04.00/KDE.Elisa.installer.yaml:9]: @stephengillie Yeah, and this is going to be a problem in long run. I am not sure about the frequency of the update. Shall I include the build number as well in the manifest?

**Comment by BrandonWanHuanSheng** [comment]
Does it have alternate url because that link is determine to be unstable, it might damage the project.

**Comment by sarvex** [comment]
@stephengillie is there a solution to this problem, it seems KDE is building the stable release everyday and updating the build version

**Comment by stephengillie** [comment]
> @stephengillie is there a solution to this problem, it seems KDE is building the stable release everyday and updating the build version

Ah - there are 2 solutions for this: 
- Locate the stable release and use that here instead. This is hte preferred solution.
- Have an automated process provide a daily update PR to the manifest.

**Comment by sarvex** [comment]
> * Locate the stable release and use that here instead. This is the preferred solution.
Tired and failed. 

> * Have an automated process provide a daily PR to update the manifest.
Is there a framework for such a script with support of declarative programming. I am coming from scoop where such a support exists

**Comment by stephengillie** [comment]
> > * Have an automated process provide a daily PR to update the manifest.
>
>   Is there a framework for such a script with support of declarative programming. I am coming from scoop where such a support exists

A couple of implementations exist - Wingetbot checks many packages daily, to ensure the package is still available and that the hash still matches. Several users have written similar automation for their own scripts.
These are usually based on one of the community manifest programs, such as [YAMLCreate](https://github.com/microsoft/winget-pkgs/blob/master/doc/tools/YamlCreate.md), [Wingetcreate](https://github.com/microsoft/winget-create), or [Komac](https://github.com/russellbanks/Komac). Some may be more suitable to scripting in Scoop or other languages than others. Or feel free to create your own manifest creation tool if you like - here is the [manifest schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0).

**Comment by sarvex** [comment]
@stephengillie there seams to a feature request under development for this #500

**Comment by stephengillie** [comment]
> @stephengillie there seams to a feature request under development for this #500

Nice find. Down the page is mentioned [WinGet Releaser](https://github.com/marketplace/actions/winget-releaser), which might be an automation tool similar to what you're looking for.

**Comment by sarvex** [comment]
@stephengillie [WinGet Releaser](https://github.com/marketplace/actions/winget-releaser), is for project owners and required github action to commit run the github workflow. I think it is better to wait for #500 to be done

**Pull request closed without merging**